grc/core/generator/cpp_hier_block.py

```diff
@@ -124,6 +124,9 @@ def var_or_value(name):
             )
         else:
             t['make'] = '{cls}()'.format(cls=block_id)
+        # Self-connect if there aren't any ports
+        if {len(data['inputs']), len(data['outputs'])} == {0}:
```

Comment:
IMO this does not convey your intend very well. May I suggest (for readability) to use  
`if not data["inputs"] and not data["outputs"]:`

Reply:
Agreed! I'll fix it

Reply:
When you do, can you please rebase onto the latest master?